code/modules/reagents/chemistry/reagents/other_reagents.dm

```diff
@@ -218,54 +218,73 @@
 
 /datum/reagent/water/holywater/on_mob_life(mob/living/carbon/M)
-	if(M.blood_volume)
-		M.blood_volume += 0.1 // water is good for you!
-	if(!data)
-		data = list("misc" = 1)
-	data["misc"]++
-	M.jitteriness = min(M.jitteriness+4,10)
+	. = ..() //includes the increased blood (re)generation rate from normal water
+
+	if(M.jitteriness < 10)
+		M.jitteriness = min(M.jitteriness+4,10)
+
 	if(iscultist(M))
 		for(var/datum/action/innate/cult/blood_magic/BM in M.actions)
 			to_chat(M, "<span class='cultlarge'>Your blood rites falter as holy water scours your body!</span>")
 			for(var/datum/action/innate/cult/blood_spell/BS in BM.spells)
 				qdel(BS)
-	if(data["misc"] >= 25)		// 10 units, 45 seconds @ metabolism 0.4 units & tick rate 1.8 sec
+
+	if(HAS_TRAIT(M, TRAIT_CULT_GHOST)) //it would be really awkward if summoned cult ghosts could be deconverted
+		exposed_mob.adjustFireLoss(5, 0) //and it burns, burns, burns~
+		return
+	
+	if(current_cycle >= 25 && current_cycle < 60)		// ~10u or more of holy water is required to reach this point
 		if(!M.stuttering)
 			M.stuttering = 1
 		M.stuttering = min(M.stuttering+4, 10)
 		M.Dizzy(5)
 		if(iscultist(M) && prob(20))
-			M.say(pick("Av'te Nar'Sie","Pa'lid Mors","INO INO ORA ANA","SAT ANA!","Daim'niodeis Arc'iai Le'eones","R'ge Na'sie","Diabo us Vo'iscum","Eld' Mon Nobis"), forced = "holy water")
-			if(prob(10))
+			M.say(pick("Av'te Nar'Sie","Pa'lid Mors","INO INO ORA ANA","SAT ANA!","Daim'niodeis Arc'iai Le'eones","R'ge Nar'sie","Diabo us Vo'iscum","Eld' Mon Nobis"), forced = "holy water")
+			if(prob(20) && !seizure) //4% chance/tick for Nar'Sie to roast you
 				M.visible_message("<span class='danger'>[M] starts having a seizure!</span>", "<span class='userdanger'>You have a seizure!</span>")
-				M.Unconscious(120)
-				to_chat(M, "<span class='cultlarge'>[pick("Your blood is your bond - you are nothing without it", "Do not forget your place", \
-				"All that power, and you still fail?", "If you cannot scour this poison, I shall scour your meager life!")].</span>")
-	if(data["misc"] >= 60)	// 30 units, 135 seconds
+				seizure = TRUE
+				M.Paralyze(60)
+				if(M.jitteriness < 40)
+					M.jitteriness = min(M.jitteriness+25,40)
+				to_chat(M, "<span class='cultlarge'>[pick("Your blood is your bond - you are NOTHING without it.", "Do not forget your place, servant.", \
+				"I gave you all that power, and you still managed to fail me? Pathetic.", "I would normally be furious at you for this, but your continued inability to perform basic tasks that I had previously believed were idiot-proof quite honestly amuses me at this point.", "You'll be back.", "If you cannot scour this poison, I shall scour your meager life, insect!", "Just have a cultist use a ritual dagger on you, you imbecile!", "I gave you magic, friends, a PURPOSE! What have THEY ever done for you?!")].</span>")
```

Comment:
Should we put these strings somewhere else? Or at least separate them by line breaks?